azure-pipelines-pr-validation.yml

```diff
@@ -42,238 +46,296 @@ variables:
-        mergeTestResults: true
-        testRunTitle: 'Unit Tests'
-      condition: and(succeededOrFailed(), ne(variables['SkipTests'], 'true'))
-
-    # Publishes setup VSIXes to a drop.
-    # Note: The insertion tool looks for the display name of this task in the logs.
-    - task: ms-vseng.MicroBuildTasks.4305a8de-ba66-4d8b-b2d1-0dc4ecbbf5e8.MicroBuildUploadVstsDropFolder@1
-      displayName: Upload VSTS Drop
-      inputs:
-        DropName: $(VisualStudio.DropName)
-        DropFolder: 'artifacts\VSSetup\$(BuildConfiguration)\Insertion'
-        AccessToken: $(_DevDivDropAccessToken)
-      condition: succeeded()
-
-    # Publish insertion packages to CoreXT store.
-    - task: NuGetCommand@2
-      displayName: Publish CoreXT Packages
-      inputs:
-        command: push
-        packagesToPush: '$(Build.SourcesDirectory)\artifacts\VSSetup\$(BuildConfiguration)\DevDivPackages\**\*.nupkg'
-        allowPackageConflicts: true
-        feedsToUse: config
-        publishVstsFeed: '97a41293-2972-4f48-8c0e-05493ae82010'
-      condition: succeeded()
-
-    # Publish an artifact that the RoslynInsertionTool is able to find by its name.
-    - task: PublishBuildArtifacts@1
-      displayName: Publish Artifact VSSetup
-      inputs:
-        PathtoPublish: 'artifacts\VSSetup\$(BuildConfiguration)'
-        ArtifactName: 'VSSetup'
-      condition: succeeded()
-
-    - task: ms-vseng.MicroBuildTasks.521a94ea-9e68-468a-8167-6dcf361ea776.MicroBuildCleanup@1
-      displayName: Perform Cleanup Tasks
-      condition: succeededOrFailed()
-
-- stage: insert
-  displayName: Create Insertion
-  dependsOn:
-    - build
-
-  jobs:
-  - job: insert
-    displayName: Insert to VS
-    pool:
-      vmImage: windows-2019
-    steps:
-    - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
-      displayName: Setting SourceBranchName variable
-      condition: succeeded()
-
-    - template: eng/pipelines/insert.yml
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+
+    - stage: build
+      displayName: Build and Test
+
+      jobs:
+      - job: PRValidationBuild
+        displayName: PR Validation Build
+        timeoutInMinutes: 360
+        templateContext:
+          outputs:
+
+          # Publish OptProf configuration files to the artifact service
+          - output: artifactsDrop
+            displayName: 'OptProf - Publish to Artifact Services - ProfilingInputs'
+            condition: succeeded()
+            dropServiceURI: 'https://devdiv.artifacts.visualstudio.com'
+            buildNumber: 'ProfilingInputs/$(System.TeamProject)/$(Build.Repository.Name)/$(SourceBranchName)/$(OriginalBuildNumber)'
+            sourcePath: '$(Build.SourcesDirectory)\artifacts\OptProf\$(BuildConfiguration)\Data'
+            toLowerCase: false
+            usePat: true
+            accessToken: $(_DevDivDropAccessToken)
+            retentionDays: 90
```

Comment:
So currently our official build doesn't use the artifactsDrop output and uses an ArtifactsDrop task instead. Do we want to move it over to match?

Reply:
Hmm, fine with this way if it works